src/sampling.jl

```diff
@@ -0,0 +1,177 @@
+module Sampling
+
+export ListSampler, RejectionSampler, Samples
+export construct_sampler, accept!
+
+using Random
+using DataStructures
+
+"""
+Struct to hold the results of a simulation.
+"""
+struct Samples{T}
+    bitstrings::DefaultDict{String, Int}
+    amplitudes::Dict{String, T}
+end
+
+Samples() = Samples(DefaultDict{String, Int}(0), Dict{String, ComplexF32}())
+
+"""Abstract type for samplers"""
+abstract type AbstractSampler end
+
+###############################################################################
+# ListSampler
+###############################################################################
+
+"""
+A Sampler struct to compute the amplitudes for a list of bitstrings.
+"""
+struct ListSampler <: AbstractSampler
+    list::Vector{String}
+    num_amplitudes::Int64
+end
+
+function ListSampler(params)
```

Comment:
Can be good to pass these as kwargs to make it more useful as interface. For dict can always use splatting (...) to expand a dict